connman/src/timezone.c

```diff
@@ -336,7 +426,7 @@ int __connman_timezone_change(const char *zone)
 		return -EIO;
 	}
 
-	err = write_file(map, &st, ETC_LOCALTIME);
+	err = write_file(map, &st, connman_setting_get_string("Localtime"));
```

Comment:
What is going on in here?
To me it looks like it reads data from /usr/share/zoneinfo/xxx and then writes it to where ever /etc/localtime or similar symlink happens to point to i.e. potentially corrupts some random zoneinfo file?

Reply:
This is part of the [clock.c D-Bus API that can change timezone](https://github.com/sailfishos/connman/blob/master/connman/src/clock.c#L288). Maybe we'd at least need a config option to disable that `__connman_timezone_change()`?

Reply:
Hmm, on closer look: if destination path is a symlink, write_file() makes an attempt to remove it before writing. So, it should not corrupt files possible symlinks point to. But it still would break the symlink chain for which we have built-in expectations here and there - mainly that tail end of realpath of /etc/localtime = name of the timezone, possibly time/date settings & timed itself.

Having an option to disable this op from dbus api might be nice. But as this is something that already has been there for ages, it could be a separate task too.

Reply:
Yes, it works exactly like that. Only think I was also worried was that it will break our timed set localtime. We can surely do that later as this D-Bus call is not apparently used as of now.